### ISSUE TITLE
Change of boundary character class in newer XeTeX and gloss-french

### ISSUE THREAD

**Issue by eg9** [reutenauer/polyglossia#145]
Due to the recent increase in the number of character classes, the boundary class is now 4095 instead of 255.

`gloss-french.ldf` contains four references to class 255, that should (conditionally) be changed. Probably the best place for the following code is `polyglossia.sty`

```
\@ifundefined{e@alloc@intercharclass@top}
  {\chardef\xpg@boundaryclass=\@cclv}
  {\let\xpg@boundaryclass=\e@alloc@intercharclass@top}
```

and then the four references to `255` in `gloss-french.ldf` should become `\xpg@boundaryclass`

The test above will return true (and the boundaryclass set to 255) if the LaTeX kernel is old; otherwise, the LaTeX kernel already takes care of setting the correct number based on the XeTeX version.

I didn't check the whole Polyglossia tree, the change might impact other language definition files.

Reference: http://tex.stackexchange.com/a/307672/4427 (the example file produces the expected result with TL 2015, but not with TL 2016/pretest)

**Comment by maieul** [comment]
cf https://github.com/josephwright/csquotes/issues/8

should it be possible to publish this change, which make french polyglossia incomptaible with TeXLive2016

**Comment by maieul** [comment]
@eg9 In this MWE

`````
\documentclass{article}
\begin{document}
\makeatletter
\meaning\e@alloc@intercharclass@top
\makeatother
\end{document}
`````

`\e@alloc@intercharclass@top` is undefined. Are your sure of you?

**Comment by eg9** [comment]
@maieul 

Unfortunately, we have to wait until LaTeX2e <2016/03/31> Patch Level 1 is rolled in.

**Comment by maieul** [comment]
ok

**Comment by maieul** [comment]
hi @reutenauer do you think that could be solved in next time? it is a major regression

**Comment by maieul** [comment]
hi @reutenauer any new ?

**Comment by maieul** [comment]
hi @reutenauer are your still available? this bug seems me importan, as it's break ascendant compatibility.

**Comment by reutenauer** [comment]
I suppose I can release a quick-and-dirty fix, but you should know better than to push for this kind of things.

**Comment by maieul** [comment]
ok, I will do a pull request

Le 7 sept. 2016 � 00:30, Arthur Reutenauer <notifications@github.com> a �crit :

**Comment by aurelberra** [comment]
Unless I have missed something, we still don't have a solution. The importance for LaTeX documents in French is obvious. Is it such a difficult problem to tackle?

**Comment by maieul** [comment]
For now, I have adapted the code in my local repository. I was waiting for #141 before publishing it. I will open a pull request quickly.

**Comment by ArchangeGabriel** [comment]
I’ve been using polyglossia when I switched to XeLaTeX from LaTeX, because it was advised to do so. But after some time and issues rising with polyglossia (like this one), I switched back to Babel which had added support for XeLaTeX during this time, and I have not found any issue with it.

Are there still reason to use polyglossia rather than Babel? Especially with the maintainer seemingly MIA? I know they are still reason to use it for RTL or non-latin script languages (which should even be fixed in 3.10, although this release is two years late now…), but everyone active here seems to be using it for french, so…

**Comment by maieul** [comment]
We use it for french AND other language… Personnaly, I switch many time between french and greek…

**Comment by jfbu** [comment]
Please fix this. It means xelatex builds for French documentations is broken for 18 months now!

**Comment by jfbu** [comment]
Any idea when merged https://github.com/reutenauer/polyglossia/pull/178 will make it into CTAN?

**Comment by maieul** [comment]
I ask since many time to push it on CTAN, but I have no news
Le dimanche 03 décembre 2017 à 14:42 +0000, Jean-François B. a écrit :

**Comment by jfbu** [comment]
@maieul Thanks. At first I did not realize #178 had fixed this issue #145 because the issue is still "Open".

**Comment by jspitz** [comment]
This has been fixed by #178.

Someone who can please close this ticket (thanks, Arthur, BTW).
